c10/cuda/CUDAStream.cpp

```diff
@@ -369,6 +378,18 @@ void setCurrentCUDAStream(CUDAStream stream) {
   current_streams[ptr->device_index] = ptr;
 }
 
+CUDAStream registerCustomCUDAStream(DeviceIndex device_index, CUDAStreamProvider stream_provider) {
+  initCUDAStreamsOnce();
+  // counter++
+  uint32_t stream_id = custom_counter++;
```

Comment:
you probably want to check that the id doesn't go over the fixed range? (kStreamsPerPool)